### DISCUSSION TITLE
Adonis 5: TypeScript issue when using related() in Model's method.

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#2217] [Help] [answered]
The following code works (I get the expected results). But Typescript keeps saying that:

> Argument of type 'string' is not assignable to parameter of type 'ExtractModelRelations<this>'.ts(2345)

```javascript
// ...

  public async getListIds() {
    const lists = await this.related('lists').query().select('id')

    return lists.map((list) => list.id)
  }

// ...
```

<img width="896" alt="Screenshot 2021-02-04 at 18 31 04" src="https://user-images.githubusercontent.com/1658644/106931778-3b326800-6717-11eb-903e-805400824a5e.png">

Is this a bug or should we proceed another way?

**Comment by thetutlage** [comment, marked as answer]
Its an issue with Typescript. Answered here earlier https://github.com/adonisjs/core/discussions/1555#discussioncomment-61505
